modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,296 @@
+Copyright (C) 2015, OpenCV Foundation, all rights reserved.
+Copyright (C) 2015, Itseez Inc., all rights reserved.
+Third party copyrights are property of their respective owners.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+  * Redistributions of source code must retain the above copyright notice,
+    this list of conditions and the following disclaimer.
+
+  * Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+
+  * Neither the names of the copyright holders nor the names of the contributors
+    may be used to endorse or promote products derived from this software
+    without specific prior written permission.
+
+This software is provided by the copyright holders and contributors "as is" and
+any express or implied warranties, including, but not limited to, the implied
+warranties of merchantability and fitness for a particular purpose are disclaimed.
+In no event shall copyright holders or contributors be liable for any direct,
+indirect, incidental, special, exemplary, or consequential damages
+(including, but not limited to, procurement of substitute goods or services;
+loss of use, data, or profits; or business interruption) however caused
+and on any theory of liability, whether in contract, strict liability,
+or tort (including negligence or otherwise) arising in any way out of
+the use of this software, even if advised of the possibility of such damage.
+*/
+
+#include "opencv2/face.hpp"
+
+using namespace std;
+
+namespace cv{
+namespace face{
+
+FacemarkKazemi::~FacemarkKazemi(){}
+FacemarkKazemiImpl:: ~FacemarkKazemiImpl(){}
+int FacemarkKazemiImpl:: getNumLandmarks() const {return numlandmarks;}
+int FacemarkKazemiImpl:: getNumFaces() const {return numfaces;}
+unsigned long FacemarkKazemiImpl::left(unsigned long index){
+    return 2*index+1;
+}
+unsigned long FacemarkKazemiImpl::right(unsigned long index){
+    return 2*index+2;
+}
+FacemarkKazemiImpl :: FacemarkKazemiImpl(CascadeClassifier face_cascade){
+    face_cascade_= face_cascade;
+    //initialise other variables
+    numfaces =1;
+    numlandmarks =194;
+    //These variables are used for training data
+    //These are initialised as described in the research paper
+    //referenced above
+    cascade_depth = 10;
+    tree_depth = 4;
+    num_trees_per_cascade_level = 500;
+    learning_rate = float(0.1);
+    oversampling_amount = 20;
+    num_test_coordinates = 400;
+    lambda = float(0.1);
+    num_test_splits = 20;
+    minmeanx=8000.0;
+    maxmeanx=0.0;
+    minmeany=8000.0;
+    maxmeany=0.0;
+}
+bool FacemarkKazemiImpl:: getFaces(Mat src,vector<Rect> &facep,double scaleFactor, int minNeighbors, int flags, Size minSize, Size maxSize){
```

Comment:
remove all spaces between namespaces and functions. It is not needed at all.
add this in the whole cpp file.